SME.CDEP.Aplicacao/Servicos/ServicoIdioma.cs

```diff
@@ -0,0 +1,50 @@
+﻿using AutoMapper;
+using SME.CDEP.Aplicacao.DTOS;
+using SME.CDEP.Aplicacao.Servicos.Interface;
+using SME.CDEP.Dominio.Entidades;
+using SME.CDEP.Infra.Dados.Repositorios.Interfaces;
+
+namespace SME.CDEP.Aplicacao.Servicos
+{
+    public class ServicoIdioma : IServicoIdioma
+    {
+        private readonly IRepositorioIdioma repositorioIdioma;
+        private readonly IMapper mapper;
+        
+        public ServicoIdioma(IRepositorioIdioma repositorioIdioma, IMapper mapper) 
+        {
+            this.repositorioIdioma = repositorioIdioma ?? throw new ArgumentNullException(nameof(repositorioIdioma));
+            this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+        public async Task<long> Inserir(IdiomaDTO idiomaDTO)
+        {
+            var idioma = mapper.Map<Idioma>(idiomaDTO);
+            return await repositorioIdioma.Inserir(idioma);
+        }
+
+        public async Task<IList<IdiomaDTO>> ObterTodos()
+        {
+            return (await repositorioIdioma.ObterTodos()).ToList().Where(w=> !w.Excluido).Select(s=> mapper.Map<IdiomaDTO>(s)).ToList();
```

Comment:
o primeiro .ToList() não faz sentido pois o where e select geram um IEnumerable novamente.

```suggestion
        public async Task<IList<IdiomaDTO>> ObterTodos()
        {
            return (await repositorioIdioma.ObterTodos()).Where(w=> !w.Excluido).Select(s=> mapper.Map<IdiomaDTO>(s)).ToList();
```